### ISSUE TITLE
wiredep and the main property of bower.json

### ISSUE THREAD

**Issue by seyDoggy** [sytelus/CryptoJS#1]
As noted here in [this issue](https://github.com/middleout/angular-cryptography/issues/5), cryptojslib defines `**/*.js` in the main property. When grunt-wiredep comes across this, it parse it into your html like so:

``` html
<script src="bower_components/cryptojslib/**/*.js"></script>
```

Any package that lists this library as a dependency will ultimately have this same issue. If you can't point to a specific file in your `main` then I suggest note defining the main property at all. That way, grunt-wiredep will throw a warning, stating that it was unable to wire up said dependency and to wire it up manually. It's better to see that warning than to have `**/*.js` added to the HTML.

**Comment by rbecheras** [comment]
You're right

**Comment by duchuy** [comment]
I have a related issue when doing unit-testing with bower, grunt and karma.
when run grunt test. wiredep will auto inject the files into
karma.conf.js

'bower_components/cryptojslib/components/aes-min.js',
'bower_components/cryptojslib/components/aes.js',
'bower_components/cryptojslib/components/cipher-core-min.js',
....
'bower_components/cryptojslib/rollups/aes.js',
'bower_components/cryptojslib/rollups/hmac-md5.js',
'bower_components/cryptojslib/rollups/hmac-ripemd160.js',
'bower_components/cryptojslib/rollups/hmac-sha1.js',
....


my bower.json
"devDependencies": {
	"cryptojslib": "^3.1.2"
},

so when I run unit testing I will get these error

PhantomJS 2.1.1 (Windows 7 0.0.0) ERROR
  ReferenceError: Can't find variable: CryptoJS
  at F:/project/bower_components/cryptojslib/components/aes-min.js:7


PhantomJS 2.1.1 (Windows 7 0.0.0) ERROR
  ReferenceError: Can't find variable: CryptoJS
  at F:/project/bower_components/cryptojslib/components/aes.js:9



If I change my bower.json to include crypto-js

"devDependencies": {
	"crypto-js": "^3.1.2",
	"cryptojslib": "^3.1.2"
}
I will get these errors when running grunt test

PhantomJS 2.1.1 (Windows 7 0.0.0) ERROR
  TypeError: undefined is not an object (evaluating 'q.lib')
  at F:/project/bower_components/cryptojslib/components/aes-min.js:7


PhantomJS 2.1.1 (Windows 7 0.0.0) ERROR
  TypeError: undefined is not an object (evaluating 'C.lib')
  at F:/project/bower_components/cryptojslib/components/aes.js:10


PhantomJS 2.1.1 (Windows 7 0.0.0) ERROR
  TypeError: undefined is not an object (evaluating 'CryptoJS.lib')
  at F:/project/bower_components/cryptojslib/components/cipher-core-min.js:7


PhantomJS 2.1.1 (Windows 7 0.0.0) ERROR
  TypeError: undefined is not an object (evaluating 'CryptoJS.lib')
  at F:/project/bower_components/cryptojslib/components/cipher-core.js:10



My project just need to use rollups/aes.js for encrypt/decrypt content. but I don't know how to remove components from the karma test script.

**Comment by rbecheras** [comment]
@demaniak this issue should be closed now

**Comment by demaniak** [comment]
I am currently completely out of the loop on this, maybe @seyDoggy can confirm and close his issue?

**Comment by seyDoggy** [comment]
Speaking of out of the loop. I'd have to look later.

**Comment by seyDoggy** [comment]
Looks like it was resolved in #2.
